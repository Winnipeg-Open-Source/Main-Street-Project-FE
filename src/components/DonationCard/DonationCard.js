import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Text } from 'pcln-design-system';
import Card from 'components/Card';
import { getDonationPath } from 'constants/paths';

function DonationCard ({ id, className, title, donationDate, onClick }) {
    const onDonationClick = () => onClick(getDonationPath(id));

    const date = donationDate || 'Unknown Donation Date';

    return (
        <Card className={className} onClick={onDonationClick}>
            <Text color='text.darkest' fontSize={1} bold>{title}</Text>
            <Text color='text.darkest' fontSize={1}>Donated: {date}</Text>
        </Card>
    );
}

DonationCard.displayName = 'DonationCard';

DonationCard.propTypes = {
    id: PropTypes.number,
    className: PropTypes.string,
    title: PropTypes.string,
    donationDate: PropTypes.string,
    onClick: PropTypes.func,
}

DonationCard.defaultProps = {
    className: '',
};

export default memo(DonationCard);
