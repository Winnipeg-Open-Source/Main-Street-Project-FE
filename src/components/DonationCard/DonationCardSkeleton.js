import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Skeleton from 'react-loading-skeleton';
import Card from 'components/Card';

const StyledCard = styled(Card)`
    display: flex;
    justify-content: space-between;
    align-items: center;
`;

function DonationCardSkeleton ({ className }) {
    return (
        <StyledCard className={className} row>
            <Skeleton width='150px' height={29} />
            <Skeleton width='100px' height={29} />
        </StyledCard>
    );
}

DonationCardSkeleton.displayName = 'DonationCardSkeleton';

DonationCardSkeleton.propTypes = {
    className: PropTypes.string,
};

DonationCardSkeleton.defaultProps = {
    className: '',
};

export default memo(DonationCardSkeleton);
