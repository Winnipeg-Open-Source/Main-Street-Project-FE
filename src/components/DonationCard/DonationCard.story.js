import React from 'react';
import DonationCard from './DonationCard';

export const WithoutDonationDate = () => <DonationCard title='Donation Title' />;

export const WithDonationDate = () => (
    <DonationCard title='Donation Title' donationDate='2020 Feb 02' />
);

export default {
    title: 'DonationCard',
    component: DonationCard,
};
