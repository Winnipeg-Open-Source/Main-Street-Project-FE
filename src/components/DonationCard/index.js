import WithSkeleton from 'hocs/WithSkeleton';
import DonationCard from './DonationCard';
import DonationCardSkeleton from './DonationCardSkeleton';

export default WithSkeleton(DonationCardSkeleton, DonationCard);
