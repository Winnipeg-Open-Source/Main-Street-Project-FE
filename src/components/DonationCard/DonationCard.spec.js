import React from 'react';
import renderWithThemeAndRouter from 'tests/helpers/renderWithThemeAndRouter';
import DonationCard from './DonationCard';

describe('DonationCard', () => {
    it('renders without crashing', () => {
        const { asFragment } = renderWithThemeAndRouter(<DonationCard />);

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders with a donation date', () => {
        const { getByText } = renderWithThemeAndRouter(
            <DonationCard donationDate='2020 February 20' />
        );

        expect(getByText('Donated: 2020 February 20')).toBeInTheDocument();
    });
});
