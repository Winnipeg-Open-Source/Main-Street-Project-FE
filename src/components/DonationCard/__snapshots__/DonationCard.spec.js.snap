// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonationCard renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-pbJYR hNzbLI sc-pkfPC fQMEfn"
  >
    <div
      class="sc-AxjAm sc-qYsuA sc-psDhf dOhwaS"
      color="background.lightest"
      width="1"
    >
      <div
        class="sc-AxhCb jTwMnz"
        color="text.darkest"
        font-size="1"
      />
      <div
        class="sc-AxhCb fYdZYD"
        color="text.darkest"
        font-size="1"
      >
        Donated: Unknown Donation Date
      </div>
    </div>
  </div>
</DocumentFragment>
`;
